astropy/time/tests/test_nan.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from astropy.time import Time, TimeDelta
+import astropy.units as u
+import pytest
+
+
+FLOAT_FORMATS = [
+    'byear',
+    'cxcsec',
+    'decimalyear',
+    'gps',
+    'jd',
+    'jyear',
+    'mjd',
+    'plot_date',
+    'stardate',
+    'unix',
+    'unix_tai',
+]
+
+STRING_FORMATS = {
+    'byear_str': 'B1950.0',
+    'jyear_str': 'J2000.0',
+    'iso': '2020-01-01 20:00:00.5',
+    'isot': '2020-01-01T20:00:00.5',
+    'yday': '2020:300:20:00:05',
+}
+
+
+def test_time_creation():
+    '''Test a time scalar can be created as nan'''
+    t = Time(np.nan, format='jd')  # for now, in principle, it shouldn't matter
+    assert np.isnan(t._time.jd2)
```

Comment:
Storing NaN in jd2 is considered an internal implementation detail of masking and could (in theory) change.  So I'd recommend not testing this here.